test/translate_c.zig

```diff
@@ -2754,4 +2754,20 @@ pub fn addCases(cases: *tests.TranslateCContext) void {
         \\    return if (x > y) x else y;
         \\}
     });
+
+    cases.add("Cast from integer literals to poiter",
+        \\#define NULL ((void*)0)
+        \\#define GPIO_0_MEM_MAP ((unsigned*)0x8000)
+        \\#define GPIO_1_MEM_MAP ((unsigned*)0x8004)
+        \\#define GPIO_2_MEM_MAP ((unsigned*)0x8008)
+        \\
+    , &[_][]const u8{
+        \\pub const NULL = @intToPtr(?*c_void, 0);
+    ,
+        \\pub const GPIO_0_MEM_MAP = @intToPtr(?[*c]c_uint, 0x8000);
```

Comment:
C pointers are already nullable.
https://ziglang.org/documentation/master/#C-Pointers
>Allows address 0. On non-freestanding targets, dereferencing address 0 is safety-checked Undefined Behavior. Optional C pointers introduce another bit to keep track of null, just like ?usize. Note that creating an optional C pointer is unnecessary as one can use normal Optional Pointers.

Reply:
Ah, of course. Perhaps the optionality should be added to `transCreateNodePtrType` ?

Reply:
The void pointer special case is in `parseCSuffixOpExpr`  you should move the optionality there. `void*` should be translated as `?*c_void`.